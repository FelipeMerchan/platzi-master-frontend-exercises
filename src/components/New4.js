import React from 'react'

import './styles/New4.css'
import noticeImage from '../images/picture.png'

function New4() {
  return (
    <section className="New4">
      <h3>Lorem ipsum dolor sit amet, consectetuer adipiscing elit</h3>
      <div>
        <div>
          <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
            nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
            volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
            ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. 
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
            nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
            volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
            ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. 
          </p>
          <h5 className="text-uppercase float-left text-center">Lorem<br />Ipsum</h5>
          <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
            nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
            volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
            ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. 
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam 
            nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat 
            volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation 
            ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. exerci tation 
            ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequatexerci tation 
            ullamcorper suscipit lobortis nisl orper suscipit lobortis nisl.
          </p>
        </div>
        <figure>
          <img className="w-100" width="500" src={ noticeImage } alt="Imagen de la noticia" />
        </figure>
      </div>
    </section>
  )
}

export default New4