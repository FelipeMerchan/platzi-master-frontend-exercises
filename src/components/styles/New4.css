.New4 {
  padding: var(--space-2) 0;
}

.New4 > h3 {
  margin-bottom: var(--space);
  font-style: italic;
  font-size: 1.4rem;
}

.New4 > div:nth-of-type(1) {
  display: grid;
  grid-template-columns: 1fr 2fr;
  grid-column-gap: var(--space);
}

.New4 figure {
  max-width: 100%;
}

h5.float-left {
  font-size: .8rem;
}