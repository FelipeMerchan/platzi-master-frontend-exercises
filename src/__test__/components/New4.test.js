import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import New4 from '../../components/New4'

describe('Probar el componente <New4 />', () => {
  const new4 = shallow(<New4 />)

  test('Render del componente New4', () => {
    expect(new4.length).toEqual(1)
  })
  test('Comprobar el tÃ­tulo', () => {
    expect(new4.find('h3').text()).toEqual('Lorem ipsum dolor sit amet, consectetuer adipiscing elit')
  })
  test('Render con el className float-left', () => {
    expect(new4.find('h5').hasClass('float-left')).toBe(true)
  })
})

describe('New4 Snapshot', () => {
  test('Comprobar la interfaz de usuario de New4', () => {
    const new4 = create(<New4 />)
    expect(new4.toJSON()).toMatchSnapshot()
  })
})