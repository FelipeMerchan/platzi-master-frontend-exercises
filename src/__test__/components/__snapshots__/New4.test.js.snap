// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New4 Snapshot Comprobar la interfaz de usuario de New4 1`] = `
<section
  className="New4"
>
  <h3>
    Lorem ipsum dolor sit amet, consectetuer adipiscing elit
  </h3>
  <div>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
      </p>
      <h5
        className="text-uppercase float-left text-center"
      >
        Lorem
        <br />
        Ipsum
      </h5>
      <p>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat.
      </p>
      <p>
        Lorem ipsum dolor sit amet, consectetuer adipiscing elit, sed diam nonummy nibh euismod tincidunt ut laoreet dolore magna aliquam erat volutpat. Ut wisi enim ad minim veniam, quis nostrud exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequat. exerci tation ullamcorper suscipit lobortis nisl ut aliquip ex ea commodo consequatexerci tation ullamcorper suscipit lobortis nisl orper suscipit lobortis nisl.
      </p>
    </div>
    <figure>
      <img
        alt="Imagen de la noticia"
        className="w-100"
        src="picture.png"
        width="500"
      />
    </figure>
  </div>
</section>
`;
